architecture/builds.adoc

```diff
@@ -0,0 +1,52 @@
+= Builds
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+== Overview
+A build is a process of transforming input parameters into resulting object.
+
+== BuildConfig
+`BuildConfig` object is the definition of the entire build process. It consists of
+following elements:
+* triggers - defines policies used for automatic invoking builds
+  * github webhook - github specific webhook, specifies which repository changes should
+    invoke new build, eg. each new commit triggers a new build. This trigger
+    is specific to github API.
+  * generic webhook - generic webhook is similar to github webhook in that it
+    invokes a new build whenever it gets a notification. The difference is its payload
+    which is slightly different than github's.
+  * image change - defines a trigger which will be invoked upon availability of
+    a new image in specified registry.
+* parameters
+  * source - describes the SCM used to locate the sources
+  * strategy - describes which build type will be invoked along with build type
+    specific details
+  * output - describes the resulting image name, tag and registry to which the
+    image should be pushed
+
+
+Since the resulting object depends on the builder used for create the image we've
+defined following three strategies:
+* link:[#Docker build]
+* link:[#STI build]
+* link:[#Custom build]
+
+== Docker build
+Docker build, under the hood, invokes plain https://docs.docker.com/reference/commandline/cli/#build[docker build],
+thus it expects a repository with a `Dockerfile` and all the necessary directories.
+
+== STI build
+STI build is a replacement for the v2 like user experience, meaning the user specifies the repository
+where his project is located and a builder image, which defines the language and framework used for writing
+his application.
+
```

Comment:
...STI will then assemble a new image which runs the application defined by the source using the framework defined by the builder image.